frontend/src/app/modules/authentication/registration/registration.component.ts

```diff
@@ -9,29 +18,59 @@ import { UserRegisterDto } from '../../../models/auth/user-register-dto';
     templateUrl: './registration.component.html',
     styleUrls: ['./registration.component.sass'],
 })
-export class RegistrationComponent implements OnInit {
+export class RegistrationComponent extends BaseComponent implements OnInit {
     public registerForm: FormGroup = new FormGroup({});
 
-    // eslint-disable-next-line no-empty-function
-    constructor(private fb: FormBuilder) {
+    public showPassword = false;
+
+    public showConfirmPassword = false;
+
+    public isEmailInvalid = false;
+
+    public isUsernameInvalid = false;
+
+    constructor(
+        private fb: FormBuilder,
+        private router: Router,
+        private spinner: SpinnerService,
+        private authService: AuthService,
+    ) {
+        super();
```

Comment:
Why you need BaseComponent? And it's constructor?

Reply:
BaseComponent provides unsubcribe$ stream, that allows to unsubscribe from observable (in this case from registration) upon component destruction. Base consctuctor's call is required by eslint